### PR TITLE
fix(oas-utils): order of imported operations

### LINKED ISSUE #4088: Why i can't get proper sequence when i import JSON in Scalar desktop app
### Discussed in https://github.com/scalar/scalar/discussions/4076

<div type='discussions-op-text'>

<sup>Originally posted by **aditya05062002** December  3, 2024</sup>
Can some one help me why i get different sequence from original in scalar desktop app. In Scalar Document i get proper sequence like post, get, put, delete, search but in scalar desktop app i get sequence  like delete, get, post, put, get why that happening if i import same JSON into both of them 
Screenshot (81)
Screenshot (83)
</div>

### PR DESCRIPTION
**Pull request by antlio**
this pr fixes #4088 in order to sort method based on path VS prior method iteration as it was altering the operation order:

⊢ before / after for [galaxy](https://galaxy.scalar.com/)
<div>
image
image
</div>

### CODE REVIEW

**Review by marclave — APPROVED** [review]

File: packages/oas-utils/src/transforms/import-spec.ts [hunk 1]
```diff
@@ -254,77 +254,79 @@ export async function importSpecToWorkspace(
...
-          const keys = Object.keys(s)
-          if (keys.length) return keys[0]
-          else return []
-        })
-
-        let selectedSecuritySchemeUids: string[] = []
+    // Creates a sorted array of methods based on the path object.
+    const methods = Object.keys(path).filter((method) =>
+      requestMethods.includes(method as any),
+    )
```
Comment (amritk) [inline packages/oas-utils/src/transforms/import-spec.ts:260]:
```suggestion
    const methods = Object.keys(path).filter(isHTTPMethod)
```
we have a type guard for this! you can just move it from the client to oas-utils

File: packages/oas-utils/src/transforms/import-spec.ts [hunk 2]
```diff
@@ -254,77 +254,79 @@ export async function importSpecToWorkspace(
...
-              authentication.preferredSecurityScheme ?? '',
-            )
-              ? authentication.preferredSecurityScheme
-              : securityRequirements[0]
-          const uid = securitySchemeMap[name]
-          selectedSecuritySchemeUids = [uid]
-        }
+    methods.forEach((method) => {
+      const operation = path[method as keyof typeof path]
+      const operationServers = serverSchema
```
Comment (amritk) [inline packages/oas-utils/src/transforms/import-spec.ts:264]:
```suggestion
      const operation = path[method]
      const operationServers = serverSchema
```
now methods should be typed :)

**Review by amritk — CHANGES_REQUESTED** [review]
Just remove the any please!

File: packages/oas-utils/src/transforms/import-spec.ts (hunk 1 above)
Comment (antlio) [reply packages/oas-utils/src/transforms/import-spec.ts:260]: oooh right right! tysm

File: packages/oas-utils/src/helpers/httpMethods.ts [hunk 3]
```diff
@@ -80,3 +80,7 @@ export const getHttpMethodInfo = (methodName: string) => {
     }
   )
 }
+
+/** Type guard which takes in a string and returns true if it is in fact an HTTPMethod */
+export const isHTTPMethod = (method: string): method is RequestMethod =>
```
Comment (hanspagel) [inline packages/oas-utils/src/helpers/httpMethods.ts:85]: code style police: pls use `isHttpMethod` (not `isHTTPMethod`)

**Review by hanspagel — CHANGES_REQUESTED** [review]

File: packages/oas-utils/src/helpers/httpMethods.ts (hunk 3 above)
Comment (antlio) [reply packages/oas-utils/src/helpers/httpMethods.ts]: great call

**Pull request merged**